lib/ansible/modules/source_control/git.py

```diff
@@ -972,10 +977,10 @@ def git_version(git_path, module):
     return LooseVersion(rematch.groups()[0])
 
 
-def git_archive(git_path, module, dest, archive, archive_fmt, version):
+def git_archive(git_path, module, dest, archive, archive_fmt, archive_prefix, version):
     """ Create git archive in given source directory """
-    cmd = "%s archive --format=%s --output=%s %s" \
-          % (git_path, archive_fmt, archive, version)
+    cmd = "%s archive --format=%s --output=%s --prefix=%s %s" \
+          % (git_path, archive_fmt, archive, archive_prefix, version)
```

Comment:
It is probably best to only add the `--prefix` flag if it is specified in the module arguments.
```suggestion
        cmd = [git_path, 'archive', '--format', archive_fmt, '--output', archive, version]
        if archive_prefix is not None:
            cmd.insert(-1, '--prefix')
            cmd.insert(-1, archive_prefix)
```